Automattic/wp-calypso
# Site Profiler: Use `final_url` to get user report

Fixes https://github.com/Automattic/dotcom-forge/issues/7155

## Proposed Changes

* Get `final_url` from the call to `/site-profiler/metrics/basic` endpoint
* Use that URL to call the `/site-profiler/lead` endpoint

## Why are these changes being made?


* There was an error when requesting the user form described in the [parent task](https://github.com/Automattic/dotcom-forge/issues/7155)

## Testing Instructions



* Apply this patch to your environment
* Go to `site-profiler` and check a URL that redirects to some other URL. For example, request a site without the  `www` that will redirect to `www`. An example could be `yahoo.es` or `netfilx.com`
* Click on the CTA `Get full site report -...` and populate the form with valid values
* Check the form is correctly sent and you can see a `Success` message
* Optionally, inspect the values in Network panel in Developer Tools for `/site-profiler/metrics/basic` and `/site-profiler/lead` to check that the URL used to confirm that the `url` and `final_url` are different in the  `/site-profiler/metrics/basic` endpoint and the correct `final_url` has been used as the payload for the `/site-profiler/lead`  endpoint

## Pre-merge Checklist
